pkg/controller/bootstrap/bootstrap.go

```diff
@@ -376,3 +395,107 @@ func parseManifests(filename string, r io.Reader) ([]manifest, error) {
+
+		// Validate annotation presence
+		if !hasPreBuiltImage || preBuiltImage == "" {
+			if isPostSeeding {
+				// Post-seeding: annotation was removed after seeding completed, this is OK, skip
+				klog.V(4).Infof("Skipping MachineOSConfig %s - no pre-built image annotation but seeding already completed", mosc.Name)
+				continue
+			}
+			// Pre-seeding: annotation is REQUIRED for install-time OCL
+			validationErrors = append(validationErrors, fmt.Errorf("MachineOSConfig %q is missing required annotation %q - this is required for install-time OCL",
+				mosc.Name, buildconstants.PreBuiltImageAnnotationKey))
+			continue
+		}
+
+		poolName := mosc.Spec.MachineConfigPool.Name
+		// verify that the pool used in the MOSC actually exists
+		poolExists := false
+		for _, pool := range pools {
+			if pool.Name == poolName {
+				poolExists = true
+				break
+			}
+		}
+		if !poolExists {
+			validationErrors = append(validationErrors, fmt.Errorf("MachineOSConfig %q references non-existent pool %q", mosc.Name, poolName))
+			continue
+		}
+
+		// Validate the pre-built image format and digest
+		if err := validatePreBuiltImage(preBuiltImage); err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("invalid pre-built image %q for MachineOSConfig %q (pool %q): %w",
+				preBuiltImage, mosc.Name, poolName, err))
+			continue
+		}
+
+		// Create the component MachineConfig
+		mc := ctrlcommon.CreatePreBuiltImageMachineConfig(poolName, preBuiltImage, buildconstants.PreBuiltImageAnnotationKey)
+		preBuiltImageMCs = append(preBuiltImageMCs, mc)
+		klog.Infof("Validated and created component MachineConfig %s with OSImageURL: %s for pool %s", mc.Name, preBuiltImage, poolName)
+	}
+
+	// Return all validation errors if any occurred
+	if len(validationErrors) > 0 {
+		return nil, fmt.Errorf("validation failed for MachineOSConfigs: %s", errors.Join(validationErrors...))
+	}
+
+	return preBuiltImageMCs, nil
+}
+
+// validatePreBuiltImage validates the pre-built image format using containers/image library
+func validatePreBuiltImage(imageSpec string) error {
+	if imageSpec == "" {
+		return fmt.Errorf("pre-built image spec cannot be empty")
+	}
+
+	// Use the containers/image library to parse and validate the image reference
+	ref, err := reference.ParseNamed(imageSpec)
+	if err != nil {
+		return fmt.Errorf("pre-built image has invalid format: %w", err)
+	}
+
+	// Ensure the reference has a digest (is canonical)
+	canonical, ok := ref.(reference.Canonical)
+	if !ok {
+		return fmt.Errorf("pre-built image must use digested format (image@sha256:digest), got: %q", imageSpec)
+	}
+
+	// Validate the digest using the go-digest library
+	if err := canonical.Digest().Validate(); err != nil {
+		return fmt.Errorf("pre-built image has invalid digest: %w", err)
+	}
+
+	// Ensure it's specifically a SHA256 digest (which is what we expect for container images)
+	if canonical.Digest().Algorithm() != digest.SHA256 {
```

Comment:
thought (non-blocking): We'll need to adjust this in future once all of the container runtimes, et. al. adopt post-quantum crypto algorithms.

(To be clear, there's nothing to do here right now.)